lib/AST/Module.cpp

```diff
@@ -2535,6 +2546,20 @@ RestrictedImportKind SourceFile::getRestrictedImportKind(const ModuleDecl *modul
   if (imports.isImportedBy(module, getParentModule()))
     return RestrictedImportKind::None;
 
+  if (importKind == RestrictedImportKind::Implicit &&
+      module->getLibraryLevel() == LibraryLevel::API) {
```

Comment:
Is there a reason to restrict this to `LibraryLevel::API`? I thought the majority of cases where this would help are actually `LibraryLevel::SPI`.

Reply:
I guess we should not implicitly import SPI modules in the public swiftinterfaces of API modules. However, I think that this would address more broken interfaces if it also handled references to implicitly imported SPI modules in SPI decls.

Reply:
I can see this being complex to get right, though, and it might not be immediately necessary.

Reply:
I restricted it to API to avoid breaking swiftinterfaces by inserting public imports of SPI modules. Yeah, that's too restrictive. I'll also allow adding any import when the source is part of a non API module. This would be in line with the current diagnostics about "public import of a private module from a public module".

We could also consider always adding the missing import if this remains a common issue. This would be more appropriate if we decided to make this as an error before Swift 6.